src/main/java/org/elasticsearch/cluster/metadata/MappingMetaData.java

```diff
@@ -354,7 +354,7 @@ private void initMappers(Map<String, Object> withoutType) {
                     path = fieldNode.toString();
                 } else if (fieldName.equals("format")) {
                     format = fieldNode.toString();
-                } else if (fieldName.equals("default")) {
+                } else if (fieldName.equals("default") && fieldNode != null) {
```

Comment:
btw -in the case where you inject a record without provided *path* property being there, and have set default=null, you'll properly get this error, as expected per the documentation quoted above.

```
{
    "error": "TimestampParsingException[failed to parse timestamp [timestamp is required by mapping]]",
    "status": 500
}
```